File: java/VendingMachine/src/com/ravendarque/credit/Credit.java
package com.ravendarque.credit;

public class Credit {

    private double totalValue;

    public double getValue() {

        return totalValue;
    }

    public void add(double value) {

        totalValue += value;
    }

    public void spend(double value)
            throws InsufficientCreditException {

        if (!validateSpend(value))
            throw new InsufficientCreditException();

        totalValue -= value;
    }

    public boolean validateSpend(double amount) {

        return totalValue >= amount;
    }
}

File: C++/VendingMachine/VendingMachine/Credit.cpp
#include "pch.h"
#include "Credit.h"
#include "InsufficientCreditError.h"

double Credit::GetValue() const
{
	return TotalValue;
}

void Credit::Add(const double value)
{
	TotalValue += value;;
}

void Credit::Spend(const double amount)
{
	if (amount > TotalValue)
		throw InsufficientCreditError();

	TotalValue -= amount;
}

bool Credit::ValidateSpend(const double amount) const
{
	return amount <= TotalValue;
}
